tests/load-tests/evaluate.py

```diff
@@ -127,8 +127,14 @@ def main():
         stats[m]["pass"]["when"] = count_stats_when(stats_raw[m]["pass"]["when"])
         stats[m]["fail"]["when"] = count_stats_when(stats_raw[m]["fail"]["when"])
 
-        if "mean" in stats[m]["pass"]["duration"]:
-            kpi_sum += stats[m]["pass"]["duration"]["mean"]
+        if stats[m]["pass"]["duration"]["samples"] == 0:
+            # If we had 0 measurements in some metric, that means not a single
+            # build made it through all steps, so kpi_sum metric does not make
+            # sense as it would only cover part of the journey
+            kpi_sum = None
```

Comment:
Just another thought on the approach. Instead of setting the **kpi_sum** to _None_, we could set it to **-1** and then have a check in the max-concurrency [branching](https://github.com/konflux-ci/e2e-tests/blob/74a1b0ebf699a4f8f7dff7d438fecac18b9c3639/tests/load-tests/run-max-concurrency.sh#L209) to break out of the execution loop as our test has failed due to reaching the limits.

Reply:
You are right. I was thinking it is OK if it fails if 0 of 1 build fails, but it is probably possible this can happen for bigger concurrency and that might be a valid result. Used `-1`.

Reply:
Yep. Just additional change is that we need to add this -1 check over [here](https://github.com/konflux-ci/e2e-tests/blob/74a1b0ebf699a4f8f7dff7d438fecac18b9c3639/tests/load-tests/run-max-concurrency.sh#L209): 

```
            elif awk "BEGIN { exit !($workloadKPI == -1 || $workloadKPI > $threshold_sec || $workloadKPIerrors / $t * 100 > $threshold_err)}"; then
```